publication-commons/src/main/java/no/unit/nva/publication/model/business/PublishingRequestCase.java

```diff
@@ -82,6 +85,39 @@ public static PublishingRequestCase fromPublication(Publication publication) {
         return openingCaseObject;
     }
 
+    public static PublishingRequestCase create(Resource resource, UserInstance userInstance,
+                                               PublishingWorkflow workflow) {
+        var publishingRequestCase = new PublishingRequestCase();
+        publishingRequestCase.setIdentifier(SortableIdentifier.next());
+        publishingRequestCase.setCustomerId(resource.getCustomerId());
+        publishingRequestCase.setStatus(TicketStatus.PENDING);
+        publishingRequestCase.setViewedBy(ViewedBy.addAll(userInstance.getUser()));
+        publishingRequestCase.setResourceIdentifier(resource.getIdentifier());
+        publishingRequestCase.setOwnerAffiliation(userInstance.getTopLevelOrgCristinId());
+        publishingRequestCase.setResponsibilityArea(userInstance.getPersonAffiliation());
+        publishingRequestCase.setOwner(userInstance.getUser());
+        publishingRequestCase.setFilesForApproval(resource.getPendingFiles());
+        publishingRequestCase.setWorkflow(workflow);
+
+        if (REGISTRATOR_PUBLISHES_METADATA_AND_FILES.equals(workflow)) {
+            publishingRequestCase.setAssignee(new Username(userInstance.getUsername()));
+            publishingRequestCase.publishApprovedFile();
+            return publishingRequestCase.complete(resource.toPublication(), userInstance);
+        }
+        if (REGISTRATOR_PUBLISHES_METADATA_ONLY.equals(workflow) && publishingRequestCase.getFilesForApproval().isEmpty()) {
+            return publishingRequestCase.complete(resource.toPublication(), userInstance);
+        } else {
+            return publishingRequestCase;
+        }
```

Comment:
Extract this to a method called `handleMetdataOnlyWorkflow`. `return REGISTRATOR_PUBLISHES_METADATA_AND_FILES.equals(workflow) ? handleMetadataAndFilesWorkflow() : handleMetdataOnlyWorkflow()`. Although I don't like the wording with `handle`